### ISSUE TITLE
As queued event listener

### ISSUE THREAD

**Issue by pktharindu** [lorisleiva/laravel-actions#87]
Thank you for this amazing package!

I'm using some actions as listeners and after switching to v2, I'm unable to queue them.

In v1, I did something like this:

```php
class MyClass extends Action implements ShouldQueue
{
    public function handle(User $user)
    {
        // ...
    }
}
```

With v2 I'm trying to do:

```php
class MyClass implements ShouldQueue
{
    use AsAction;

    public function handle(User $user)
    {
        // ...
    }

    public function asListener($event)
    {
        $this->handle($event->user);
    }
}
```

But it throws the following error.

```
Argument 1 passed to App\Actions\MyClass::handle() must be an instance of App\Models\User, instance of App\Events\MyEvent given,
```

But if I remove the `ShouldQueue` interface, it works just fine.

What I'm I doing wrong and how can I make this work?

**Comment by lorisleiva** [comment]
Hi there 👋 

Thanks for raising this. I'm super busy with another project at the moment but I'll take a look at this as soon as I can.

**Comment by FrankVisuals** [comment]
Having the same issue here - do you know when you'll be able to change this?

**Comment by lorisleiva** [comment]
Hi 👋 

I don't I'm afraid but for now you can dispatch the action as a Job from the `asListener` method and it will be queued.

**Comment by davidstoker** [comment]
I ran into this as well in starting to use v2 of this package and adjusted my approach to make it work. @pktharindu or @FrankVisuals, how are you registering your listeners?

I have all of my events and listeners centrally registered in the `EventServiceProvider`:

```
class EventServiceProvider extends ServiceProvider
{
    /**
     * The event listener mappings for the application.
     *
     * @var array
     */
    protected $listen = [
        ExampleEvent::class => [
            ExampleAction::class
        ]
    ];
}
```

By default, Laravel's Dispatcher uses `handle` but it also supports callable style for listeners. 

https://github.com/laravel/framework/blob/de5f6a7ac9be7674820d24c1d5e9ababcaf00c3e/src/Illuminate/Events/Dispatcher.php#L449

I changed my listener using this package to explicitly specify `asListener` and it will queue and process correctly:

```
class EventServiceProvider extends ServiceProvider
{
    /**
     * The event listener mappings for the application.
     *
     * @var array
     */
    protected $listen = [
        ExampleEvent::class => [
            [ExampleAction::class, 'asListener'],
        ]
    ];
}
```

@lorisleiva Thanks for making this package by the way! Very helpful and test mocks are very handy.

**Comment by pktharindu** [comment]
@davidstoker I decided to stick with the v1 for the time being. But this sure looks like a good workaround.
Maybe I'll have another go at it. 

Anyway, thanks for sharing!

**Comment by lorisleiva** [comment]
Hi there 👋 

This has been fixed and released as [`v2.1.1`](https://github.com/lorisleiva/laravel-actions/releases/tag/v2.1.1). 🚀 

Let me know if that all works fine for you. Thanks again for raising this! 🍺
